felles/src/main/kotlin/no/nav/helsearbeidsgiver/felles/domene/Forespoersel.kt

```diff
@@ -23,39 +23,42 @@ data class Forespoersel(
     val forespurtData: ForespurtData,
     val erBesvart: Boolean,
 ) {
-    fun forslagBestemmendeFravaersdag(): LocalDate =
-        finnTidligste(
-            spleisForslag = bestemmendeFravaersdager[orgnr],
-            beregnet =
-                bestemmendeFravaersdag(
-                    arbeidsgiverperioder = emptyList(),
-                    sykefravaersperioder = sykmeldingsperioder + egenmeldingsperioder,
-                ),
-        )
-
-    fun forslagInntektsdato(): LocalDate =
-        finnTidligste(
-            spleisForslag = bestemmendeFravaersdager.minOfOrNull { it.value },
-            beregnet =
-                bestemmendeFravaersdag(
-                    arbeidsgiverperioder = emptyList(),
-                    sykefravaersperioder = sykmeldingsperioder + egenmeldingsperioder,
-                ),
-        )
+    fun forslagBestemmendeFravaersdag(): LocalDate {
+        val forslag = bestemmendeFravaersdager[orgnr]
+        return brukForslagEllerUtled(forslag)
+    }
+
+    fun forslagInntektsdato(): LocalDate {
+        val forslag = bestemmendeFravaersdager.minOfOrNull { it.value }
+        return brukForslagEllerUtled(forslag)
+    }
 
     fun eksternBestemmendeFravaersdag(): LocalDate? =
         bestemmendeFravaersdager.minus(orgnr).minOfOrNull {
             it.value
         }
 
-    private fun finnTidligste(
-        spleisForslag: LocalDate?,
-        beregnet: LocalDate,
-    ): LocalDate = listOfNotNull(spleisForslag, beregnet).min()
+    private fun brukForslagEllerUtled(forslag: LocalDate?): LocalDate {
+        val utledet =
+            bestemmendeFravaersdag(
+                arbeidsgiverperioder = emptyList(),
+                sykefravaersperioder = egenmeldingsperioder.plus(sykmeldingsperioder).sortedBy { it.fom },
```

Comment:
Ser du har lagt på en sortering her. Er rekkefølgen avgjørende for logikken, eller er det mest for at det skal være lettere å lese ved debugging og slikt når periodene ligger i riktig rekkefølge?

Reply:
Rekkefølgen er viktig for logikken. `bestemmendeFravaersdag`-funksjonen henter den første i `sykefravaersperioder` og anser det som starten av sykefraværet.